examples/components/utils/aggregatemodelweights/run.py

```diff
@@ -65,7 +65,9 @@ def add_model(self, model_path: str):
         """
         if self.avg_state_dict is None:
             # no model yet, nothing to average
-            self.avg_state_dict = torch.load(model_path)
+            self.avg_state_dict = torch.load(
+                model_path, map_location=torch.device("cpu")
```

Comment:
Please forgive my ignorance - can you explain what this does, why it is required, and what are the implications if someone wants to run this on a gpu for some reason? 

I'm sure I'm missing something, but it does look like we're hardcoding the machine type. Hence my concern.

Reply:
By default PyTorch tries to deserialize model on the device it was trained on and thus if the model was trained on CUDA device but there is no CUDA available on the cluster used for aggregating weights, this will fail. More info here: https://stackoverflow.com/questions/67934005/pytorch-cpu-cuda-device-load-without-gpu 

I have updated it to not to be hardcoded, even though it would always work, to always used cuda if available otherwise use the cpu. I hope this is nicer solution :)